File: Ch02/Python/DataSciencePrg01.py
import statistics

''' DataSciencePrg01 Sample program to calculate mean median mode 
standard deviation and variance '''

class DataSciencePrg01:
    # Initialize method to initialize all the variable
    def __init__(self):
        self.example_list = [5, 2, 5, 6, 1, 2, 6, 7, 2, 6, 3, 5, 5]
        self.x = 0
        self.y = 0
        self.z = 0
        self.a = 0
        self.b = 0

    # Method or function used to calculate Statistical Mean
    def mean(self):
        self.x = statistics.mean(self.example_list)
        print(self.x)

    # Method or function  used to calculate Statistical median
    def median(self):
        self.y = statistics.median(self.example_list)
        print(self.y)

    # Method or function  used to calculate Statistical mode
    def mode(self):
        self.z = statistics.mode(self.example_list)
        print(self.z)

    # Method or function  used to calculate Statistical stdev
    def stdev(self):
        self.a = statistics.stdev(self.example_list)
        print(self.a)

    # Method or function  used to calculate Statistical variance
    def variance(self):
        self.b = statistics.variance(self.example_list)
        print(self.b)

# Main method to run all methods for testing
if __name__ == '__main__':
    dataScience = DataSciencePrg01()
    dataScience.mean()
    dataScience.median()
    dataScience.mode()
    dataScience.stdev()
    dataScience.variance()

File: Ch02/devops4datascience/src/main/java/com/packt/devops/datascience/DataSciencePrg01.java
package com.packt.devops.datascience;


import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;

public class DataSciencePrg01 {
    public static double x = 0;
    public static double y = 0;
    public static double z = 0;
    public static double a = 0;
    public static double b = 0;
    public static int intArray[] = { 5, 2, 5, 6, 1, 2, 6, 7, 2, 6, 3, 5, 5 };
    public static DescriptiveStatistics stats = null;

    /**
     * Main Method for DataSciencePrg01
     *
     * @param args
     */
    public static void main(String[] args) {
        // Integer array for the program

        DataSciencePrg01 dataScience = new DataSciencePrg01();
        dataScience.init();
        dataScience.mean();
        dataScience.median();
        dataScience.mode();
        dataScience.stdev();
        dataScience.variance();

    }

    public static void init() {
        // Get a SummaryStatistics instance
        stats = new DescriptiveStatistics();
        for (int i : intArray)
            stats.addValue(i);
    }

    /**
     *
     * Method used to calculate mean
     */
    public static void mean() {
        x = stats.getMean();
        System.out.println(x);
    }

    /**
     *
     * Method used to calculate median
     */
    public static void median() {
        y = stats.getPercentile(50);
        System.out.println(y);

    }

    /**
     *
     * Method used to calculate mode
     */
    public static void mode() {

        int maxValue = 0, maxCount = 0;

        for (int i = 0; i < intArray.length; ++i) {
            int count = 0;
            for (int j = 0; j < intArray.length; ++j) {
                if (intArray[j] == intArray[i])
                    ++count;
            }
            if (count > maxCount) {
                maxCount = count;
                maxValue = intArray[i];
            }
        }

        z = maxValue;
        System.out.println(z);

    }

    /**
     *
     * Method used to calculate stdev
     */
    public static void stdev() {
        a = stats.getStandardDeviation();
        System.out.println(a);

    }

    /**
     *
     * Method used to calculate stdev
     */
    public static void variance() {
        b = stats.getVariance();
        System.out.println(b);
    }

}
